config/opal_check_cuda.m4

```diff
@@ -50,9 +50,18 @@ AC_ARG_WITH([cuda],
             [AS_HELP_STRING([--with-cuda(=DIR)],
             [Build cuda support, optionally adding DIR/include])])
 AC_MSG_CHECKING([if --with-cuda is set])
+
+# Search for libcuda.so in $with_cuda if the user didn't pass --with-cuda-libdir
+# Otherwise check for cuda in the default path, /usr/local/cuda. If the default
+# path doesn't exist, set with_cuda_libdir to empty.
 AC_ARG_WITH([cuda-libdir],
             [AS_HELP_STRING([--with-cuda-libdir=DIR],
-                            [Search for CUDA libraries in DIR])])
+                            [Search for CUDA libraries in DIR])],
+            [],
+            [AS_IF([test -d "$with_cuda"],
+             [with_cuda_libdir=$(dirname $(find $with_cuda -name libcuda.so) 2> /dev/null)],
+             [with_cuda_libdir=$(dirname $(find /usr/local/cuda -name libcuda.so) 2> /dev/null)])
```

Comment:
If `/usr/local/cuda` does not exist, `find` will emit an error.

Reply:
good point

Reply:
Isn't that ok though? The error will make it explicit that /usr/local/cuda is where the configure script is looking.

Reply:
The error won't be captured in `config.log`, which is where such things should be recorded.  The `configure` output generally shouldn't contain errors from commands.  If it does, it's generally a mistake that we should fix.

Reply:
I updated the find command to redirect to /dev/null in https://github.com/open-mpi/ompi/pull/12385. There are AC_WARN prints that should take care of letting the user know /usr/local/cuda is where the script is checking.